test/with_dummyserver/test_connection.py

```diff
@@ -33,6 +34,29 @@ def test_returns_urllib3_HTTPResponse(pool: HTTPConnectionPool) -> None:
     assert isinstance(response, HTTPResponse)
 
 
+@pytest.mark.skipif(not hasattr(sys, "audit"), reason="requires python 3.8")
+def test_audit_event(pool: HTTPConnectionPool) -> None:
+    audit_event = None
+    audit_args: tuple[typing.Any, ...] = (None, None, None)
+
+    def hook(event: str, args: tuple[typing.Any]) -> None:
+        nonlocal audit_event, audit_args
+        if event == "http.client.connect":
+            audit_event = event
+            audit_args = args
+
+    sys.addaudithook(hook)  # type: ignore
+
+    conn = pool._get_conn()
+    conn.request("GET", "/")
+
+    assert audit_event == "http.client.connect"
+    assert len(audit_args) > 2
+    assert audit_args[0] == conn
+    assert audit_args[1] == conn.host
+    assert audit_args[2] == conn.port
```

Comment:
It's a pity that the audit hook cannot be temporary in this case. Changing the test to check that the `audit` function is called instead of adding a hook accessing nonlocal variables from all next tests seems to fix the segmentation faults.

```suggestion
@pytest.mark.skipif(not hasattr(sys, "audit"), reason="requires python 3.8")
@patch("urllib3.connection.sys.audit")
def test_audit_event(audit_mock: Mock, pool: HTTPConnectionPool) -> None:
    conn = pool._get_conn()
    conn.request("GET", "/")
    audit_mock.assert_any_call("http.client.connect", conn, conn.host, conn.port)
    # Ensure the event is raised only once.
    connect_events = [
        call for call in audit_mock.mock_calls if call.args[0] == "http.client.connect"
    ]
    assert len(connect_events) == 1
```